contracts/Verifier.sol

```diff
@@ -36,7 +36,9 @@ contract Verifier is Ownable, HydratedRuntime {
         address challengerAddr;
 
         bytes32 solverPath;
+        uint256 solverDepth;
         bytes32 challengerPath;
+        uint256 challengerDepth;
```

Comment:
I know it's still WIP but we only need one property to track the depth, right?

Reply:
I put it there because there is the case when `solverDepth > challengerDepth` which I haven't found a convenient solution.
When `challengerDepth > solverDepth`, challenger can just challenge with part of his tree. But it is not applicable in the other case. I have considered challenger make a fake tree with zero on the right node, but I think it's awkward :thinking:

Reply:
@peara Yes, good point. The challenger resize his tree in this case, either by shrinking or by adding zero leaves. This works while we take the left path if the right one is all zero 👍.
I think shrinking is already implemented in the unit tests. But that issue is for another day to come 😄

Reply:
@pinkiebell However, there is a problem. When solver put a arbitrary high value to `solverDepth`, something like 10 billions. Challenger would have no legitimate way to generate the corresponding tree. Even if it is possible, the dispute would never finish on time. 

What do you think? Should we put a cap on that value? Or else, we need to add `challengerDepth`, so after the dispute is initialized, the burden of proof such tree will be on the solver.

Reply:
> However, there is a problem. When solver put a arbitrary high value to solverDepth, something like 10 billions. Challenger would have no legitimate way to generate the corresponding tree. Even if it is possible, the dispute would never finish on time.

In this case the solver would lose anyway if the challenger submit his part in a round :)

But cap'ing the execution length makes much sense. We already talked about it on some meetings but postponed the decision for a later time.

I see it like this:
maxExecutionDepth = 2**12  = a max. of 16.777.216 execution steps

> Or else, we need to add challengerDepth, so after the dispute is initialized, the burden of proof such tree will be on the solver.

If the solver registers such big tree, why is the burden on the solver if we are going to track challengerDepth ? There still need to be round-based setup. 🤔

Reply:
> > In this case the solver would lose anyway if the challenger submit his part in a round :)
> 
> I think the solver can prepare such tree beforehand and can submit an endless array of meaningless hashes. So, the dispute would never reach a leaf.

And that's why we have a claimTimeout function :)

> 
> > If the solver registers such big tree, why is the burden on the solver if we are going to track challengerDepth ? There still need to be round-based setup.
> 
> In my opinion, the one with the bigger tree need to prove that they have such tree in the first place. For example, using a function to submit a partial of the tree. After this, both trees will have the same height.

The whole point of that merkle tree path-walking is that it makes it possible to prove on ethereum mainnet. It would be too costly If we are even trying to proof a partial set of any given tree each.

Reply:
In `claimTimeout` 
https://github.com/leapdao/solEVM-enforcer/blob/master/contracts/Verifier.sol#L287-L291
```
bool solverWins = true;
if ((dispute.state & CHALLENGER_VERIFIED) != 0 || (dispute.state & CHALLENGER_RESPONDED) != 0) {
            solverWins = false;
}
```
It's not clear for me how solver would be slashed if he register something like `executionHash: fakeHash, executionDepth: 100000`.
Challenger could be the one who would get slashed.

> The whole point of that merkle tree path-walking is that it makes it possible to prove on ethereum mainnet. It would be too costly If we are even trying to proof a partial set of any given tree each.

I see. The function to validate event a partial tree could well beyond the current gas limit. So I guess a cap would be best. The above case would also be controllable with a cap.

There is another thing I would like to ask. We are currently not refresh the timeout of dispute, so one party can just wait until the end of dispute period then submit `respond`. For example with the current `claimTimeout`, if solver does that, right after `respond`, `updateRound` will run and both `SOLVER_RESPONDED` and `CHALLENGER_RESPONDED` would be reset and the challenger will be slashed by `claimTimeout`.

Reply:
> There is another thing I would like to ask. We are currently not refresh the timeout of dispute, so one party can just wait until the end of dispute period then submit respond. For example with the current claimTimeout, if solver does that, right after respond, updateRound will run and both SOLVER_RESPONDED and CHALLENGER_RESPONDED would be reset and the challenger will be slashed by claimTimeout.

Good point. The reverse could also happen (for solver). One problem with refreshing the dispute timeout every time is obvsly a long-tail timing attack. And we also have to take care of the whole challengePeriod of the Enforcer contract 🤔 

One solution for this problem might be to disallow `respond/updateRound` for a time `dispute.timeout - RESPOND_GRACE_PERIOD` before we reach the dispute timeout. To give the opponent time to handle.
-OR-
Better: Force submission to `respond` to happen in a specific time frame, so that the solver or challenger automatically get slashed if they did not respond in time.